app/ui-utils/client/lib/RoomHistoryManager.js

```diff
@@ -119,6 +124,37 @@ export const RoomHistoryManager = new class {
 		return this.histories[rid];
 	}
 
+	async queue() {
+		return new Promise((resolve) => {
+			const requestId = uuidv4();
+			const done = () => {
+				this.lastRequest = new Date();
+				resolve();
+			};
+			if (this.requestsList.length === 0) {
+				return this.run(done);
+			}
+			this.requestsList.push(requestId);
+			this.once(requestId, done);
+		});
+	}
+
+	run(fn) {
+		const difference = differenceInMilliseconds(new Date(), this.lastRequest);
+		if (!this.lastRequest || difference > 400) {
+			return fn();
+		}
+		return setTimeout(fn, 400 - difference);
```

Comment:
in my tests using `500` I had no errors.
```suggestion
		if (!this.lastRequest || difference > 500) {
			return fn();
		}
		return setTimeout(fn, 500 - difference);
```